cmd/snap/cmd_snap_op.go

```diff
@@ -414,10 +414,15 @@ func (x *cmdInstall) installOne(name string, opts *client.SnapOptions) error {
 	} else {
 		changeID, err = cli.Install(name, opts)
 	}
-	if e, ok := err.(*client.Error); ok && e.Kind == client.ErrorKindSnapAlreadyInstalled {
-		fmt.Fprintf(Stderr, i18n.G("snap %q is already installed, see \"snap refresh --help\"\n"), name)
+	if e, ok := err.(*client.Error); ok {
+		msg, isError := clientErrorToCmdMessage(name, e)
```

Comment:
I've done this refactor, although it doesn't seem cleaner nor clearer to me now but I'm the one writing it so :-)